api/python/quilt3/main.py

```diff
@@ -41,6 +54,31 @@ def cmd_catalog():
         command += ["-e", var]
     command += ["-p", "3000:80", "quiltdata/catalog"]
     subprocess.Popen(command)
+
+def _launch_local_s3proxy():
+    """"
+    Launches an s3 proxy (via docker)
+    on localhost:5002
+    """
+    command = ["docker", "run", "--rm"]
+    command += ["-p", "5002:80", "quiltdata/s3proxy"]
+    subprocess.Popen(command)
+
+def cmd_catalog():
+    """
+    Run the Quilt catalog locally
+    """
+
+    local_catalog_url = "http://localhost:3000"
+    local_s3proxy_url = "http://localhost:5002"
+
+    if not _test_url(local_catalog_url):
+        _launch_local_catalog()
+
+    if not _test_url(local_s3proxy_url):
+        _launch_local_s3proxy()
+
+    open_url(local_catalog_url)
```

Comment:
[Optional]

I would add a check before `open_url` to see if both `_test_url(local_catalog_url)` and `_test_url(local_s3proxy_url)` pass. I had a situation where my docker daemon wasn't running so `quilt3 catalog` opened a "cannot connect" webpage. It was easy to go back to my terminal and see the cause, but IMO it would have been better had the webpage not opened.

Reply:
Because the catalog and s3 proxy processes run in the background, we don't know exactly when the processes stabilize or fail. I'd rather avoid adding complex retry logic here, at least for the first version.